aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/UpdateHandler.java

```diff
@@ -27,11 +33,11 @@
         final CallbackContext callbackContext,
         final Logger logger) {
 
-        // Everything except RetentionPolicyInDays and KmsKeyId is createOnly
         final ResourceModel model = request.getDesiredResourceState();
         final ResourceModel previousModel = request.getPreviousResourceState();
         final boolean retentionChanged = ! retentionUnchanged(previousModel, model);
         final boolean kmsKeyChanged = ! kmsKeyUnchanged(previousModel, model);
+        final boolean tagsChanged = ! tagsUnchanged(previousModel, model);
```

Comment:
Should use `getDesiredResourceTags` which will combine both stack tags and resource tag.  And `getPreviousResourceTags` for both previous stack tags and resource tags. 

You will need to override the `resourceDefinedTags` ([example](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kms/blob/master/key/src/main/java/software/amazon/kms/key/Configuration.java#L20)) so that the `getDesiredResourceTags` will merge the stacks and resource defined tags together.

Reply:
Good point, makes sense and will try to add it.